### DISCUSSION TITLE
Like the robust implementation but my workflow requires listing associated session windows

### DISCUSSION THREAD

**Discussion by zach-is-my-name** [joshmedeski/sesh#115] [Ideas]
Specifically I run tmux without a status-bar (mostly) for improved focus.  I'd be able to switch if the I could list windows associated with the sessions as well.  I came upon sesh because I had to migrate my dot files to a new machine and for some reason my original binding wasn't working and I sought an alternative while debugging.  Here's the my current working binding and a view of what it does.  I'd much prefer sesh with gum though. 

```
bind -n M-m display-popup -h 70% -E 'bash -c "\\
current_session=\$(tmux display-message -p \"#S\"); \\
selected=\$(tmux list-sessions -F \"#{session_name}:\" \\
| sed \"/^\${current_session}:/d\" \\
| xargs -I{} tmux list-windows -t {} -F \"#{session_name}:#{window_index} - #{window_name}\" \\
| fzf --reverse --cycle --bind alt-j:down,alt-k:up,esc:abort \\
| sed -E \"s/\\s-.*$//\"); \\
session_name=\$(echo \"\${selected}\" | cut -d \":\" -f 1); \\
window_index=\$(echo \"\${selected}\" | cut -d \":\" -f 2 | cut -d \" \" -f 1); \\
tmux switch-client -t \"\${session_name}:\${window_index}\" \\
"'
```
<img width="1168" alt="Screenshot 2024-05-21 at 3 45 50 PM" src="https://github.com/joshmedeski/sesh/assets/16063092/129d59fb-9622-4ed6-b05a-64a486b55a6a">

**Comment by joshmedeski** [comment]
It wouldn't work in my workflow since all my window names are symbols, but I could see others benefiting from it.

When I launch v2 I don't mind creating a ticket and assigning it to someone in the community.

**Reply by zach-is-my-name** [reply]
Okay I'll track it and give it a star.  Just to put it on your radar, I'm pretty content with what I have.. I remember last time I searched there weren't many tmux switchers out there.  Seeing a dedicated project like is really good.

**Reply by joshmedeski** [reply]
Check out my friend @omerxx's sessionx plugin, it might better fit your needs:
https://github.com/omerxx/tmux-sessionx

**Reply by zach-is-my-name** [reply]
Thanks for the tip. Followed you here and over on YouTube. On the subject of fuzzy finding sessions without creating another thread, do you have any knowledge of a command that could be looped in to grep (rg) across the running sessions return the pid and switch to it? THAT would be the holy grail for me.

**Reply by joshmedeski** [reply]
Check out `man tmux`. The `FORMATS` describes variables that can be used to query tmux, including PIDs. Using grep and a fuzzy matcher should each have their own man pages to help break down the features.

Tools like ChatGPT have been helpful for me as well when trying to build new, complicated, CLI commands. Good luck!
